src/Shared/CertificateGeneration/WindowsCertificateManager.cs

```diff
@@ -10,6 +10,11 @@ internal class WindowsCertificateManager : CertificateManager
     {
         private const int UserCancelledErrorCode = 1223;
 
+        // Cache to store the result of validating whether a key is exportable or not.
+        // This avoid requesting the key multiple times and avoids an issue when we try to access
+        // the key after it has been exported.
+        private Dictionary<string, bool> _areKeysExportable = new Dictionary<string, bool>();
```

Comment:
What prevents this dictionary from unbounded growth?

Reply:
> If we get [dotnet/runtime#36287](https://github.com/dotnet/runtime/pull/36287) into preview5 (not sure why that currently targets 'master'), would this change be needed?

From our last tactics discussion dotnet/runtime were planning to do this after preview5. Given that this seems to be happening in preview5, we can pause on this PR.

Reply:
I just confirmed with the runtime team - they'll be taking the fix later in 5.0. So this is still needed!

Reply:
And it seems I've been overruled by the higher powers and the runtime fix will be brought into 5.0-preview5!